adlfs/core.py

```diff
@@ -252,18 +290,32 @@ class AzureBlobFileSystem(AbstractFileSystem):
 
     def __init__(
         self,
-        account_name: str,
         container_name: str,
-        account_key: str,
-        custom_domain: str = None,
-        is_emulated: bool = False,
+        account_name=None,
+        account_key=None,
+        sas_token=None,
+        is_emulated=False,
+        protocol=DEFAULT_PROTOCOL,
+        endpoint_suffix=SERVICE_HOST_BASE,
+        custom_domain=None,
+        request_session=None,
+        connection_string=None,
+        socket_timeout=None,
+        token_credential=None,
     ):
         AbstractFileSystem.__init__(self)
```

Comment:
It's potentially important to pass through all arguments to `super().__init__()`. The instance caching is based on these arguments.

Reply:
Doesn't that happen before init is called at all? I would say, that the signature should contain **kwargs, and we pass those on.

Reply:
Ah, yes you're right. `AbstractFileSystem.__init__` doesn't actually use args or kwargs anymore.